includes/class-content-distribution.php

```diff
@@ -0,0 +1,422 @@
+<?php
+/**
+ * Newspack Network Content Distribution.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Network;
+
+use Newspack\Data_Events;
+use WP_Post;
+use WP_Error;
+
+/**
+ * Main class for content distribution
+ */
+class Content_Distribution {
+
+	/**
+	 * The post meta key for the distributed post configuration.
+	 */
+	const DISTRIBUTED_POST_META = 'newspack_network_distributed';
+
+	/**
+	 * Post meta key for the linked post containing the distributed post hash.
+	 */
+	const POST_HASH_META = 'newspack_network_post_hash';
+
+	/**
+	 * Post meta key for the linked post containing the distributed post full payload.
+	 */
+	const POST_PAYLOAD_META = 'newspack_network_post_payload';
+
+	/**
+	 * Post meta key for the linked post to determine whether the post is unlinked.
+	 */
+	const POST_UNLINKED_META = 'newspack_network_post_unlinked';
+
+	/**
+	 * Initialize this class and register hooks
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		add_action( 'init', [ __CLASS__, 'register_listeners' ] );
+		add_filter( 'newspack_webhooks_request_priority', [ __CLASS__, 'webhooks_request_priority' ], 10, 2 );
```

Comment:
This requires https://github.com/Automattic/newspack-plugin/pull/3587, which is currently up for review.